handlers/tembachat/handler_test.go

```diff
@@ -0,0 +1,77 @@
+package tembachat
+
+import (
+	"net/http/httptest"
+	"testing"
+
+	"github.com/nyaruka/courier"
+	. "github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/courier/test"
+)
+
+var testChannels = []courier.Channel{
+	test.NewMockChannel("8eb23e93-5ecb-45ba-b726-3b064e0c56ab", "TWC", "", "", nil),
+}
+
+var handleTestCases = []IncomingTestCase{
+	{
+		Label:                "Receive Valid Message",
+		URL:                  "/c/twc/8eb23e93-5ecb-45ba-b726-3b064e0c56ab/receive",
+		Data:                 `{"type": "message", "message": {"identifier": "65vbbDAQCdPdEWlEhDGy4utO", "text": "Join"}}`,
+		ExpectedRespStatus:   200,
+		ExpectedBodyContains: "Accepted",
+		ExpectedMsgText:      Sp("Join"),
+		ExpectedURN:          "webchat:65vbbDAQCdPdEWlEhDGy4utO",
+	},
+	{
+		Label:                "Invalid URN",
+		URL:                  "/c/twc/8eb23e93-5ecb-45ba-b726-3b064e0c56ab/receive",
+		Data:                 `{"type": "message", "message": {"identifier": "xxxxx", "text": "Join"}}`,
+		ExpectedRespStatus:   400,
+		ExpectedBodyContains: "invalid webchat id: xxxxx",
+	},
+	{
+		Label:                "Missing fields",
+		URL:                  "/c/twc/8eb23e93-5ecb-45ba-b726-3b064e0c56ab/receive",
+		Data:                 `{"foo": "message"}`,
+		ExpectedRespStatus:   400,
+		ExpectedBodyContains: "Field validation for 'Type' failed on the 'required' tag",
+	},
+}
+
+func TestIncoming(t *testing.T) {
+	RunIncomingTestCases(t, testChannels, newHandler(), handleTestCases)
+}
+
+func setSendURL(s *httptest.Server, h courier.ChannelHandler, c courier.Channel, m courier.MsgOut) {
+	defaultSendURL = s.URL
```

Comment:
I think it i better to set the config send URL here and remove `defaultSendURL`

Reply:
I'm just copying how other handlers are done but I did wonder if all handlers should support `send_url` in config and then this can be the normal way of overriding it in tests

Reply:
that could work too